pkg/clients/eks/eks.go

```diff
@@ -19,702 +19,335 @@ package eks
+				in.EncryptionConfig[i].Provider = v1beta1.Provider{
+					KeyArn: *e.Provider.KeyArn,
+				}
+			}
+		}
 	}
-
-	ec2Filters := []ec2.Filter{}
-	for name, values := range filters {
-		ec2Filters = append(ec2Filters, ec2.Filter{Name: aws.String(name), Values: values})
+	if in.Logging == nil && cluster.Logging != nil && len(cluster.Logging.ClusterLogging) > 0 {
+		setup := make([]v1beta1.LogSetup, len(cluster.Logging.ClusterLogging))
+		for i, cl := range cluster.Logging.ClusterLogging {
+			types := make([]v1beta1.LogType, len(cl.Types))
+			for j, t := range cl.Types {
+				types[j] = v1beta1.LogType(t)
+			}
+			setup[i] = v1beta1.LogSetup{
+				Enabled: cl.Enabled,
+				Types:   types,
+			}
+		}
+		in.Logging = &v1beta1.Logging{
+			ClusterLogging: setup,
+		}
 	}
+	if cluster.ResourcesVpcConfig != nil {
+		in.ResourcesVpcConfig.EndpointPrivateAccess = awsclients.LateInitializeBoolPtr(in.ResourcesVpcConfig.EndpointPrivateAccess, cluster.ResourcesVpcConfig.EndpointPrivateAccess)
+		in.ResourcesVpcConfig.EndpointPublicAccess = awsclients.LateInitializeBoolPtr(in.ResourcesVpcConfig.EndpointPublicAccess, cluster.ResourcesVpcConfig.EndpointPublicAccess)
+		if len(in.ResourcesVpcConfig.PublicAccessCidrs) == 0 && len(cluster.ResourcesVpcConfig.PublicAccessCidrs) > 0 {
+			in.ResourcesVpcConfig.PublicAccessCidrs = cluster.ResourcesVpcConfig.PublicAccessCidrs
+		}
 
-	request := e.amiClient.DescribeImagesRequest(&ec2.DescribeImagesInput{
-		Filters: ec2Filters,
-	})
-	out, err := request.Send(context.TODO())
-
-	if err != nil {
-		return nil, err
+		if len(in.ResourcesVpcConfig.SecurityGroupIDs) == 0 && len(cluster.ResourcesVpcConfig.SecurityGroupIds) > 0 {
+			in.ResourcesVpcConfig.SecurityGroupIDs = cluster.ResourcesVpcConfig.SecurityGroupIds
+		}
+		if len(in.ResourcesVpcConfig.SubnetIDs) == 0 && len(cluster.ResourcesVpcConfig.SubnetIds) > 0 {
+			in.ResourcesVpcConfig.SubnetIDs = cluster.ResourcesVpcConfig.SubnetIds
+		}
 	}
+	in.RoleArn = awsclients.LateInitializeStringPtr(in.RoleArn, cluster.RoleArn)
+	in.Version = awsclients.LateInitializeStringPtr(in.Version, cluster.Version)
+}
 
-	result := make([]*ec2.Image, len(out.Images))
-	for i := 0; i < len(out.Images); i++ {
-		result[i] = &out.Images[i]
+// IsUpToDate checks whether there is a change in any of the modifiable fields.
+func IsUpToDate(p *v1beta1.ClusterParameters, cluster *eks.Cluster) (bool, error) {
+	patch, err := CreatePatch(cluster, p)
+	if err != nil {
+		return false, err
 	}
 
-	return result, nil
-}
-
-func getMostRecentImage(images []*ec2.Image) *ec2.Image {
-	var result *ec2.Image
-	for i := 0; i < len(images); i++ {
-		if result == nil || aws.StringValue(result.CreationDate) < aws.StringValue(images[i].CreationDate) {
-			result = images[i]
+	// NOTE(hasheddan): AWS removes insignificant bits from CIDRs, so we must
+	// compare by converting user-supplied CIDRs to network blocks. We only skip
+	// comparison if both external and local have no CIDR blocks defined.
```

Comment:
Taking a look at `TestIsUpToDate` should paint a better picture of what is happening here.